File: 18_Bit_Manipultion/04_Reverse_bits/main.py
def reverse_bits(n):
    reversed_bits = 0
    for i in range(32):
        reversed_bits <<= 1
        reversed_bits |= n & 1
        n >>= 1
    return reversed_bits

# Example usage
n = 21  # Example input (00000000000000000000000000010101)
reversed_n = reverse_bits(n)

print(f"Original: {n}")  # 21
print(f"Reversed: {reversed_n}")  # 2818572288

File: 18_Bit_Manipultion/04_Reverse_bits/main.cpp
#include <stdio.h>
#include <stdint.h>

// Function to reverse the bits of a 32-bit unsigned integer
uint32_t reverseBits(uint32_t n) {
    uint32_t reversed = 0;
    for (int i = 0; i < 32; ++i) {
        reversed <<= 1;
        reversed |= (n & 1);
        n >>= 1;
    }
    return reversed;
}

int main() {
    uint32_t n = 21; // Example input (00000000000000000000000000010101)
    uint32_t reversed = reverseBits(n);
    
    printf("Original: %u\n", n); // 21
    printf("Reversed: %u\n", reversed); // 2818572288

    return 0;
}
